src/lm.jl

```diff
@@ -298,20 +335,23 @@ for each observation in linear model `obj`, giving an estimate of the influence
 of each data point.
 Currently only implemented for linear models without weights.
 """
+## To remove when https://github.com/JuliaStats/StatsAPI.jl/pull/16 is merged
+function crossmodelmatrix(model::RegressionModel; weighted::Bool=false) 
+    x = weighted ? modelmatrix(model; weighted=weighted) : modelmatrix(model)
+    return Symmetric(x' * x)
+end
+    
+
 function StatsBase.cooksdistance(obj::LinearModel)
-    u = residuals(obj)
-    mse = dispersion(obj,true)
+    wts = weights(obj)
+    u = residuals(obj; weighted=isweighted(obj))
+    mse = GLM.dispersion(obj,true)
     k = dof(obj)-1
     d_res = dof_residual(obj)
-    X = modelmatrix(obj)
-    XtX = crossmodelmatrix(obj)
+    X = modelmatrix(obj; weighted=isweighted(obj))
+    XtX = crossmodelmatrix(obj; weighted=isweighted(obj))
     k == size(X,2) || throw(ArgumentError("Models with collinear terms are not currently supported."))
-    wts = obj.rr.wts
-    if isempty(wts)
-        hii = diag(X * inv(XtX) * X')
-    else
-        throw(ArgumentError("Weighted models are not currently supported."))
-    end
+    hii = diag(X * inv(XtX) * X')
```

Comment:
This isn't tested currently, right?

Reply:
It should, if it isn't, it will be shortly.